rust/flatbuffers/src/vector.rs

```diff
@@ -96,84 +112,62 @@ impl<'a, T: Follow<'a> + 'a> Vector<'a, T> {
     }
 }
 
+/// Provides access to the values of a `Vector` as a slice
+///
+/// As we cannot guarantee memory alignment of the vector's data,
+/// this is only supported for types with no alignment requirements
 pub trait SafeSliceAccess {}
+
 impl<'a, T: SafeSliceAccess + 'a> Vector<'a, T> {
     pub fn safe_slice(self) -> &'a [T] {
-        let buf = self.0;
-        let loc = self.1;
-        let sz = size_of::<T>();
-        debug_assert!(sz > 0);
-        let len = unsafe { read_scalar_at::<UOffsetT>(buf, loc) } as usize;
-        let data_buf = &buf[loc + SIZE_UOFFSET..loc + SIZE_UOFFSET + len * sz];
-        let ptr = data_buf.as_ptr() as *const T;
-        let s: &'a [T] = unsafe { from_raw_parts(ptr, len) };
-        s
+        assert_eq!(align_of::<T>(), 1);
+        let data = self.data();
+        // SAFETY:
+        // Valid vector at construction time, i.e. containing element count elements of T
+        // and T has no additional alignment requirements
+        unsafe {
+            core::slice::from_raw_parts(data.as_ptr() as *const T, data.len() / size_of::<T>())
+        }
     }
 }
 
 impl SafeSliceAccess for u8 {}
 impl SafeSliceAccess for i8 {}
-impl SafeSliceAccess for bool {}
-
-// TODO(caspern): Get rid of this. Conditional compliation is unnecessary complexity.
```

Comment:
This is a fairly major change, as we can't reason about the alignment of the underlying buffer allocation, we cannot reason about the alignment of these. I opted to just remove them, but would welcome other thoughts

Reply:
I want to deprecate `SafeSliceAccess` entirely - it's not worth having two APIs so things with alignment 1 can use native slices. Your call if you want to bring that depreciation in scope of this PR. 

I think you didn't need to get rid of the impl for bool though, that should be alignment 1.

Reply:
> I want to deprecate SafeSliceAccess entirely - it's not worth having two APIs so things with alignment 1 can use native slices. Your call if you want to bring that depreciation in scope of this PR.

I think I'd like to handle this as a separate change, I think is needed for some of the codegen related to strings.

> I think you didn't need to get rid of the impl for bool though, that should be alignment 1.

Yeah, the issue with bool is not alignment but the fact it isn't trivially transmutable from an arbitrary `u8`. Rust defines bools to only have the values `0` or `1`, and so it is technically UB to transmute say `2_u8` into a bool.

Reply:
How we handled this in planus was by always returning rust-slices for `[uint8]` and `[int8]`, but using our vector type for everything else.